From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Configure uBlock Origin

Sets uBlock Origin to use a custom/enhanced config, and changes certain settings by default.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/toolkit/components/extensions/parent/ext-storage.js b/toolkit/components/extensions/parent/ext-storage.js
index 0d6109c25019c..d350ffeb608ad 100644
--- a/toolkit/components/extensions/parent/ext-storage.js
+++ b/toolkit/components/extensions/parent/ext-storage.js
@@ -10,6 +10,7 @@ ChromeUtils.defineESModuleGetters(this, {
   ExtensionStorageIDB: "resource://gre/modules/ExtensionStorageIDB.sys.mjs",
   NativeManifests: "resource://gre/modules/NativeManifests.sys.mjs",
   extensionStorageSession: "resource://gre/modules/ExtensionStorage.sys.mjs",
+  IFPrefUtils:             "moz-src:///ironfox/utils/IFPrefUtils.sys.mjs",
 });
 
 var { ExtensionError } = ExtensionUtils;
@@ -411,6 +412,46 @@ this.storage = class extends ExtensionAPIPersistent {
         managed: {
           async get(keys) {
             enforceNoTemporaryAddon(extension.id);
+            const assetsBootstrapLocation = IFPrefUtils.getStringPref("browser.ironfox.uBO.assetsBootstrapLocation");
+            const autoCommentFilterTemplate = IFPrefUtils.getStringPref("browser.ironfox.uBO.autoCommentFilterTemplate");
+            const autoUpdateDelayAfterLaunch = IFPrefUtils.getStringPref("browser.ironfox.uBO.autoUpdateDelayAfterLaunch");
+            const disableWebAssembly = IFPrefUtils.getStringPref("browser.ironfox.uBO.disableWebAssembly");
+            const filterAuthorMode = IFPrefUtils.getStringPref("browser.ironfox.uBO.filterAuthorMode");
+            const uiPopupConfig = IFPrefUtils.getStringPref("browser.ironfox.uBO.uiPopupConfig");
+            const updateAssetBypassBrowserCache = IFPrefUtils.getStringPref("browser.ironfox.uBO.updateAssetBypassBrowserCache");
+            if (extension.id == "uBlock0@raymondhill.net") {
+              return {
+                adminSettings: {
+                  assetsBootstrapLocation
+                },
+                advancedSettings: [
+                  [
+                    "autoCommentFilterTemplate",
+                    autoCommentFilterTemplate
+                  ],
+                  [
+                    "autoUpdateDelayAfterLaunch",
+                    autoUpdateDelayAfterLaunch
+                  ],
+                  [
+                    "disableWebAssembly",
+                    disableWebAssembly
+                  ],
+                  [
+                    "filterAuthorMode",
+                    filterAuthorMode
+                  ],
+                  [
+                    "uiPopupConfig",
+                    uiPopupConfig
+                  ],
+                  [
+                    "updateAssetBypassBrowserCache",
+                    updateAssetBypassBrowserCache
+                  ]
+                ]
+              }
+            }
             let data = await getManagedStorageManifestData(extension, context);
             return ExtensionStorage._filterProperties(extension.id, data, keys);
           },
-- 
